ts/src/connections/tunnelClientBase.ts

```diff
@@ -236,6 +250,36 @@ export class TunnelClientBase
         if (this.connectionProtocol === webSocketSubProtocolv2) {
             pfs.messageFactory = this;
             pfs.onForwardedPortConnecting((e) => this.onForwardedPortConnecting(e));
+            pfs.remoteForwardedPorts.onPortAdded((e) => this.onForwardedPortAdded(pfs, e));
+            pfs.remoteForwardedPorts.onPortUpdated((e) => this.onForwardedPortAdded(pfs, e));
+        }
+    }
+
+    private onForwardedPortAdded(pfs: PortForwardingService, e: ForwardedPortEventArgs) {
+        const port = e.port.remotePort;
+        if (typeof port !== 'number') {
+            return;
+        }
+
+        // If there are disconnected streams for the port, re-connect them now.
+        const disconnectedStreamsCount = this.disconnectedStreams.get(port)?.length ?? 0;
+        for (let i = 0; i < disconnectedStreamsCount; i++) {
+            pfs.connectToForwardedPort(port).catch((error) => {
+                this.trace(
```

Comment:
Is `this.traceWarning()` available here? (Just to make these log lines a little shorter).

Reply:
No. Applications just provide a single callback function for tracing, not an interface.